src/core/entity.js

```diff
@@ -84,13 +86,20 @@ class Entity {
     return this.local_matrix.slice(12, 15);
   }
 
-  set scale(vec) {
-    this._scale = vec;
-    math.mat4.scale(this.local_matrix, this.local_matrix, vec);
+  set local_scale(vec) {
+    // Scale is given in values relative to the initial (1, 1, 1) scaling.
+    // Divide vec by the current scale to get the relative scaling.  Store the
+    // result temporarily in this._local_scale to avoid creating a new vector.
+    math.vec3.divide(this._local_scale, vec, this._local_scale);
```

Comment:
This is modeled after Unity's `Transform` component: https://docs.unity3d.com/ScriptReference/Transform.html where all properties relative to the local coordinate system are prefixed with `local`. Interestingly there's no global `scale` in Unity because parents' rotations introduce skew (as you noted above). There's only `lossyScale` which I don't know yet how it's computed.